### PR TITLE
fix (fn[]) usage in hiccup

### LINKED ISSUE #15703: fix all anonymous inline function in hiccup
usually we use them wrong with `safe-area` and `:f>` components

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes: https://github.com/status-im/status-mobile/issues/15703

also `safe-area` replaced by constants, now it can be used in any place as
`(safe-area/get-insets)`

I've seen a few places where `:f> (fn []` was used two times in the same component, so I'm not sure these components will work as expected after this change cc @OmarBasem

### CODE REVIEW

**Comment by ibrkhalil** [comment]
Is it worth it creating a macro for the f-COMP function?
Personally I think it’d make it harder to miss this best practice.

Maybe to have something like
```clj
(defn my-comp [args]
(rf/create-functional-component :my-comp args))
```

**Comment by flexsurfer** [comment]
> Is it worth it creating a macro for the f-COMP function? Personally I think it’d make it harder to miss this best practice.
> 
> Maybe to have something like
> 
> ```clojure
> (defn my-comp [args]
> (rf/create-functional-component :my-comp args))
> ```

yes, that was my first thought, but I'm not sure

File: src/mocks/js_dependencies.cljs [hunk 1]
```diff
@@ -154,10 +154,7 @@ globalThis.__STATUS_MOBILE_JS_IDENTITY_PROXY__ = new Proxy({}, {get() { return (
 (def net-info #js {})
 (def touchid #js {})
 (def react-native-image-viewing #js {:default {}})
-(def safe-area-context
-  (clj->js {:SafeAreaProvider      {:_reactNativeIphoneXHelper {:getStatusBarHeight (fn [])}}
-            :SafeAreaInsetsContext {:Consumer (fn [])}
-            :SafeAreaView          {}}))
+(def safe-area-context #js {:default {}})
```
Comment (Parveshdhull) [inline src/mocks/js_dependencies.cljs]: hi @flexsurfer, thank you for the great work.
Should we rename this to `static-safe-area-insets`

**Comment by flexsurfer** [comment]
cc @J-Son89 for components test

**Review by Parveshdhull — COMMENTED** [review]
lgtm, awesome job :tada:

**Review by Parveshdhull — APPROVED** [review]

File: src/mocks/js_dependencies.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/mocks/js_dependencies.cljs]: it doesn't matter, but only just to have less question to it, i'll rename it :)

**Review by ilmotta — APPROVED** [review]
Great work, thanks @flexsurfer!

`<rant>`
It's so unfortunate that Reagent is causing this much trouble. I wish it could support functional components by default without this horrible `:f>` thing, just like newer libraries do, which were built for React hooks from the get-go, like Helix https://github.com/lilactown/helix/blob/master/docs/motivation.md. It's not the only problem in Reagent though...
`</rant>`

> Is it worth it creating a macro for the f-COMP function?

Indeed @OmarBasem, this is quite common. In Helix there is the `defnc` aka *defn component*, which takes care of normalizing some stuff to help devs.

**Comment by flexsurfer** [comment]
yeah, reagent needs some love from contributors, I feel like its poorly maintained after the author had abandoned it :(

**Comment by ilmotta** [comment]
@flexsurfer, the pattern you established here is worthy of the guidelines. Could you please add a section there?

**Comment by flexsurfer** [comment]
> @flexsurfer, the pattern you established here is worthy of the guidelines. Could you please add a section there?

its already here https://github.com/status-im/status-mobile/pull/15686/files#diff-b5e21cecfbe129756a13073f455833097d90938c8367957243f550fe8d76e567R39

**Comment by ilmotta** [comment]
> > @flexsurfer, the pattern you established here is worthy of the guidelines. Could you please add a section there?
> 
> its already here https://github.com/status-im/status-mobile/pull/15686/files#diff-b5e21cecfbe129756a13073f455833097d90938c8367957243f550fe8d76e567R39

Sorry I didn't explain correctly. I mean in this PR you named functional components with the prefix `f-`. It'd be good to have this in the guidelines since it's a strong convention. In the guidelines, the name of the function example is not following the new `f-` prefix convention.

**Comment by flexsurfer** [comment]
so, it seems like its better to use `:f>` keyword, in that case, we separate components with hooks, and it's better to move ratoms outside the `f-` component, so we have ratoms and hooks separation

**Comment by flexsurfer** [comment]
> Sorry I didn't explain correctly. I mean in this PR you named functional components with the prefix `f-`. It'd be good to have this in the guidelines since it's a strong convention. In the guidelines, the name of the function example is not following the new `f-` prefix convention.

done

**Review by ajayesivan — APPROVED** [review]

**Review by J-Son89 — APPROVED** [review]

File: src/status_im2/contexts/chat/lightbox/zoomable_image/view.cljs [hunk 2]
```diff
@@ -204,76 +204,72 @@
...
+                             (gesture/simultaneous pinch pan-x pan-y)
+                             (gesture/exclusive double-tap tap))]
+      [gesture/gesture-detector {:gesture composed-gestures}
+       [reanimated/view
+        {:style (style/container dimensions
+                                 animations
+                                 @set-full-height?
+                                 (= curr-orientation orientation/portrait))}
+        [reanimated/fast-image
+         {:source    {:uri (:image content)}
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/lightbox/zoomable_image/view.cljs]: The component was previously two functions. It now has become one function, isn't this going to cause unnecessary re-renders?

And we now have a second `let` binding in the same function, which makes it redundant, right?

File: src/status_im2/contexts/chat/lightbox/zoomable_image/view.cljs [hunk 3]
```diff
@@ -204,76 +204,72 @@
              (anim/animate-decay pan-y velocity [lower-bound upper-bound])
              (anim/animate-decay pan-y-start velocity [lower-bound upper-bound]))))))))
 
-(defn zoomable-image
+(defn f-zoomable-image
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/lightbox/zoomable_image/view.cljs:207]: same thing here

File: src/status_im2/contexts/chat/lightbox/view.cljs [hunk 4]
```diff
@@ -117,129 +117,131 @@
                           (anim/animate opacity 1)
                           (anim/animate layout 0)))))))
 
-(defn lightbox
+(defn f-lightbox
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/lightbox/view.cljs]: same thing here

**Review by OmarBasem — CHANGES_REQUESTED** [review]
I just tested the PR builds. Please roll back the parts in zoomable-image and lightbox that have nested functional components. It is not working as expected. I can refactor it in a separate issue.

**Comment by flexsurfer** [comment]
@OmarBasem done

File: src/status_im2/contexts/chat/photo_selector/view.cljs [hunk 5]
```diff
@@ -107,46 +107,48 @@
         [quo/icon (if photos? :i/chevron-down :i/chevron-up)
          {:color (colors/theme-colors colors/neutral-100 colors/white)}]]])))
 
-(defn photo-selector
+(defn f-photo-selector
   [{:keys [scroll-enabled on-scroll]}]
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/photo_selector/view.cljs]: and photo-selector please

Comment (flexsurfer) [reply src/status_im2/contexts/chat/photo_selector/view.cljs]: done

**Review by OmarBasem — APPROVED** [review]

**Pull request merged**